src/App.tsx

```diff
@@ -78,50 +82,56 @@ const useAppVersionWatcher = () => {
   }, [referencedUI, setIsAppVersionOutdated]);
 };
 
-const NetworkDropdown = lazy(() => import('./components/NetworkDropdown'));
-
 const App = () => {
   const netConfig = useAtomValue(networkConfigAtom);
   const [chainStorageWatcher, setChainStorageWatcher] = useAtom(
     chainStorageWatcherAtom,
   );
-  const [error, setError] = useState<unknown | null>(null);
-
-  const networkDropdown = import.meta.env.VITE_NETWORK_CONFIG_URL ? (
-    <></>
-  ) : (
-    <Suspense>
-      <NetworkDropdown />
-    </Suspense>
-  );
+  const { setChainConnectionError } = useStore(appStore);
+  const setRpcNode = useSetAtom(rpcNodeAtom);
 
   useEffect(() => {
     let isCancelled = false;
     if (chainStorageWatcher) return;
     const startWatching = async () => {
       try {
-        const { rpc, chainName } = await fetchChainInfo(netConfig.url);
+        const { rpc, chainName, api } = await fetchChainInfo(netConfig.url);
         if (isCancelled) return;
+        setRpcNode(rpc);
         setChainStorageWatcher(
-          makeAgoricChainStorageWatcher(rpc, chainName, e => {
+          makeAgoricChainStorageWatcher(api, chainName, e => {
             console.error(e);
-            setError(e);
-            throw e;
+            setChainConnectionError(
+              new Error(
+                'Received invalid response from API Endpoint: ' + e.message,
+              ),
+            );
           }),
         );
 
         watchVbank();
       } catch (e) {
         if (isCancelled) return;
-        setError(e);
+        setChainConnectionError(
+          new Error(
+            `Error fetching network config from ${netConfig.url}` +
```

Comment:
_question_:
Why is this using 
```
e instanceof Error ? `: ${e.message}` : ''
```
 while just above it's only  
```
e.message
```

Is it because we know `makeAgoricChainStorageWatcher` is giving us an instance of `Error` but we don't know if the `catch` is?